cvat-ui/src/utils/opencv-wrapper/intelligent-scissors.ts

```diff
@@ -118,14 +128,27 @@ export default class IntelligentScissorsImplementation implements IntelligentSci
                     state.path = state.path.slice(0, state.anchors[points.length - 1].start);
                     delete state.anchors[points.length - 1];
                 }
-
-                tool.applyImage(matImage);
-                tool.buildMap(new cv.Point(prevX, prevY));
-                tool.getContour(new cv.Point(curX, curY), contour);
-
                 const pathSegment = [];
-                for (let row = 0; row < contour.rows; row++) {
-                    pathSegment.push(contour.intAt(row, 0) + offsetX, contour.intAt(row, 1) + offsetY);
+                if (!state.blocked) {
+                    tool.applyImage(matImage);
+                    tool.buildMap(new cv.Point(prevX, prevY));
+                    tool.getContour(new cv.Point(curX, curY), contour);
+
+                    for (let row = 0; row < contour.rows; row++) {
+                        pathSegment.push(contour.intAt(row, 0) + offsetX, contour.intAt(row, 1) + offsetY);
+                    }
+                } else {
+                    const xDiff = prevX - curX;
+                    const yDiff = prevY - curY;
+                    if (xDiff !== 0 && yDiff !== 0) {
+                        const pathLength = Math.sqrt((xDiff) ** 2 + (yDiff) ** 2);
+                        const pathOffsetValue = 0.4;
+                        const pathOffsetRatio = pathOffsetValue / pathLength;
+                        pathSegment.push((curX + offsetX) + xDiff * pathOffsetRatio,
+                            (curY + offsetY) + yDiff * pathOffsetRatio);
+                    } else {
+                        pathSegment.push(curX + offsetX + 0.25, curY + offsetY + 0.25);
+                    }
```

Comment:
IMHO if the algorithm locked, it should not do anything (and probably we even does not need to run the function). To be honest I do not understand what happens here, what is ``pathOffsetValue`` and why it is 0.4, what is ``pathOffsetRatio``, what is maginc number 0.25.. 

Maybe it is necessary, but in this case need to be commented

Reply:
This code creates small offset in prev point direction to make anchors hoverable. 0.4 is absolute path offset. Then we calculate how much it takes from real path length and safe it in pathOffsetRatio. Then we apply this offset.
In case points have the same coordinates just use constant offset - 0.25. Those numbers are obtained experimentally and it seems that they work fine in most cases.
I've added some comments here

Reply:
Removed